src/prompts/password.rs

```diff
@@ -342,30 +375,47 @@ impl<'a> PasswordPrompt<'a> {
     }
 
     fn prompt<B: PasswordBackend>(mut self, backend: &mut B) -> InquireResult<String> {
-        let final_answer: String;
+        let mut first_answer = String::new();
+        let mut has_read_first_input = false;
 
-        loop {
+        let final_answer = loop {
```

Comment:
I like the structure and I like how it integrates nicely with the existing approach.

There are two paths to the UX:

- What you did, which is what I initially preferred: simply replace the main prompt in-place for the verification prompt.
- What I feel might be worth a try, after trying the first approach:  a new line.

First one feels like the better solution, but with the verification messages being the same it doesn't work as a good UX. Even with a different verification message, I feel it would require the user to actually pay attention to the prompt and they wouldn't have the option of just looking at the text and inferring everything they needed, like with the 2-line option.

Here's what I thought and let me know if you agree:

```
// First prompt:

? Password:

===========
// Second step after user submits first password:
===========

? Password: ****** (this can also be blank as is the default behavior)
  Confirmation:

===========
// Following step after user submits incorrect confirmation password
===========

# The passwords don't match.
? Password: 

===========
// Following step after user submits the password again
===========

? Password: ******
  Confirmation:

===========
// Following step after user intuitively presses ESC after realizing they wrote the first password incorrectly.
// It goes back to the first password.
// If the user presses ESC at this stage, we call `cancel_prompt!` to actually leave.
===========

? Password: 

===========
// Following step after user submits the password _again_
===========

? Password: ******
  Confirmation:

===========
// _Final render_ after confirmation is successful
// Only the first line is rendered. Just like in Date prompts where we hide the calendar and just display the final date, or Select prompts where we do the same with the lists.
===========

? Password: ******
? _next prompt_

Reply:
I am starting to be more convinced of using two lines for the password input flow instead of one, especially with the suggestion to have the ESC behavior for the second prompt. Otherwise, a suggestion I came to think about was to perhaps instead of using a custom verification/confirmation message that is shown after the main one, was to have a separate info line (akin to the error line) which tells the user to confirm the password. Not sure if there is prior art for this UX behavior in this crate (or even in other CLI tools for that matter), but it _seemed_ more consistent in having it clean up the input each time.

With that said, I think I am fine with having it as you suggested it here, since it makes it less ambiguous and allows for a nicer UX flow in the general case.

As for the details of implementing this, I assume we'll change the default confirmation message from being the same into being `"Confirmation:"`. Also, ESC on the second prompt will not trigger a Result bubbling up to the call site, but rather reset the whole flow to the first prompt being empty (or as an alternative, have the prompt being filled in from previous input in order to allow small edits of the password instead of rewriting it as a whole).